app/main/lib/shared_models/video_model.py

```diff
@@ -14,61 +15,96 @@
-                if video.hash_value and not existing.hash_value:
-                    existing.hash_value = video.hash_value
-                    flag_modified(existing, 'hash_value')
-                if video.context not in existing.context:
-                    if isinstance(video.context, list):
-                        existing.context.append(video.context[0])
-                    else:
-                        existing.context.append(video.context)
-                    flag_modified(existing, 'context')
-                saved_video = existing
-        except NoResultFound as e:
-            # Otherwise, add new video, but with context as an array
-            if video.context and not isinstance(video.context, list):
-                video.context = [video.context]
-            db.session.add(video)
-            saved_video = video
-        except Exception as e:
-            db.session.rollback()
-            raise e
-        try:
-            db.session.commit()
-            return saved_video
-        except Exception as e:
-            db.session.rollback()
-            raise e
+    def overload_context_to_denote_content_type(self, task):
+        app.logger.error(f"input to overload_context_to_denote_content_type_v2 is {task}")
+        return {**task, **{"context": {**task.get("context", {}), **{"content_type": "video"}}}}
+
+    def delete(self, task):
+        return media_crud.delete(task, Video)
+
+    def add(self, task):
+        hash_value = (task.get("result", {}) or {}).get("hash_value")
+        s3_folder = (task.get("result", {}) or {}).get("folder")
+        s3_filepath = (task.get("result", {}) or {}).get("filepath")
+        if hash_value:
+            task["hash_value"] = hash_value
+        added, obj = media_crud.add(task, Video, ["hash_value", "folder", "filepath"])
+        download_file_from_s3(s3_folder, s3_filepath, media_crud.tmk_file_path(obj.folder, obj.filepath))
+        return added
+
+    def blocking_search(self, task, modality):
+        video, temporary, context, presto_result = media_crud.get_blocked_presto_response(task, Video, modality)
+        video.hash_value = presto_result["body"]["hash_value"]
+        if video:
+            matches = self.search(task).get("result")
+            if temporary:
+                media_crud.delete(task, Video)
+            else:
+                media_crud.save(video, Video, ["hash_value"])
+            if task.get("limit"):
+                return {"result": matches[:task.get("limit")]}
+            else:
+                return {"result": matches}
+        else:
+            return {"error": "Video not found for provided task", "task": task}
 
     def async_search(self, task, modality):
         return media_crud.get_async_presto_response(task, Video, modality)
 
     def get_tempfile(self):
         return tempfile.NamedTemporaryFile()
 
-    def execute_command(self, command):
-        return os.popen(command).read()
+    def execute_command(self, cmd, params):
```

Comment:
I still don't think we should allow executing arbitrary SQL against the database.  I think is cleaner and safer to have a functions corresponding to the the actions, put the SQL inside like an ORM

Reply:
I understand and am sympathetic to this, but I don't think this is the place to address it. Perhaps lets discuss on a call some time soon to get something on the board? @computermacgyver what do you think?